packages/sanity/src/core/components/CapabilityGate.test.tsx

```
@@ -0,0 +1,59 @@
+import {render, screen} from '@testing-library/react'
+import {of} from 'rxjs'
+import {beforeEach, expect, it, vi} from 'vitest'
+
+import {createTestProvider} from '../../../test/testUtils/TestProvider'
+import {useRenderingContextStore} from '../store/_legacy/datastores'
+import {CapabilityGate} from './CapabilityGate'
+
+vi.mock('../store/_legacy/datastores.ts')
+
+beforeEach(() => {
+  vi.clearAllMocks()
+})
+
+it('renders the child if the capability is not provided by the rendering context', async () => {
+  const wrapper = await createTestProvider()
+
+  vi.mocked(useRenderingContextStore).mockReturnValue({
+    renderingContext: of({
+      name: 'default',
+      metadata: {},
+    } as const),
+    capabilities: of({}),
+  })
+
+  render(
+    <CapabilityGate capability="globalUserMenu">
+      <div data-testid="user-menu">User</div>
+    </CapabilityGate>,
+    {wrapper},
+  )
+
+  expect(screen.getByTestId('user-menu')).toBeTruthy()
```

NIT: `.toBeInTheDocument()` rather than `.toBeTruthy()`

---

Thanks. Curious: is that just a style preference?

---

Functionally, in this case there's no difference - but if you were using `queryByTestId` instead then that throws instead of returns null - so mostly around consistency of approach I suppose. `toBeTruthy/Falsy` I feel are best avoided in many cases for the additional false positives they might capture eg `""` `0` - again not relevant in this case where you are returning the element

---

Thank you for explaining. Makes sense! 🙂 I'm actually usually very opposed to truthy/falsey checks, for the reasons you stated.